libs/forecasting/src/lib/ForecastingDashboard/Controls/TransformationCreationCategorical.tsx

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  Stack,
+  Text,
+  Label,
+  ComboBox,
+  IComboBox,
+  IComboBoxOption
+} from "@fluentui/react";
+import {
+  defaultModelAssessmentContext,
+  ModelAssessmentContext
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import React from "react";
+
+import { forecastingDashboardStyles } from "../ForecastingDashboard.styles";
+import { Operation, Feature } from "../Interfaces/Transformation";
+
+export interface ITransformationCreationCategoricalProps {
+  transformationOperation?: Operation;
+  transformationFeature?: Feature;
+  transformationValue: number;
+  transformationValueErrorMessage?: string;
+  onChangeTransformationValue: (newValue: number) => void;
+}
+
+export class TransformationCreationCategorical extends React.Component<ITransformationCreationCategoricalProps> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public render(): React.ReactNode {
+    const classNames = forecastingDashboardStyles();
+    let options = undefined;
+    if (this.props.transformationFeature) {
+      options = this.context.jointDataset.metaDict[
```

Comment:
could you add a comment here ? there's a lot of indexing going on

Reply:
Makes sense. I've added a short comment.